lib/descriptors/MessageDescriptor.js

```diff
@@ -97,6 +98,18 @@ MessageDescriptor.prototype.addField = function (field) {
   return this
 }
 
+/**
+ * Adds a synthetic field to this message
+ * e.g. for use by a post-processor to add another field to a message after parsing before compiling
+ *
+ * @param {string|FieldType} type The synthetic field type, e.g. string or int32
+ * @param {string} name The field name
+ * @param {number} tag The tag number for this field
+ * @return {MessageDescriptor}
+ */
+MessageDescriptor.prototype.addSyntheticField = function (type, name, tag) {
+  return this.addField(new FieldDescriptor(type, name, tag))
+}
```

Comment:
an alternative I decided against: 
add `FieldDescriptor` to the module exports so clients could create their own and use the existing `addField(...)`.

That seemed like a big abstraction leak for little gain; plus, adding synthetic fields shouldn't be ordinary, so an explicit method seemed like a nice way to make that clear.